src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -400,7 +400,7 @@ private BoundExpression BindThrownExpression(ExpressionSyntax exprSyntax, Diagno
             else
             {
                 // In C# 8 and later we follow the ECMA specification, which neatly handles null and expressions of exception type.
-                boundExpr = GenerateConversionForAssignment(Compilation.GetWellKnownType(WellKnownType.System_Exception), boundExpr, diagnostics);
+                boundExpr = GenerateConversionForAssignment(GetWellKnownType(WellKnownType.System_Exception, diagnostics, exprSyntax), boundExpr, diagnostics);
```

Comment:
>GetWellKnownType(WellKnownType.System_Exception, diagnostics, exprSyntax) [](start = 60, length = 73)

Is this now be a breaking change for ```throw null;``` by comparison to earlier language versions?

Reply:
No, because there is a conversion for assignment from `null` to `System.Exception`.

---
In reply to: [354033064](https://github.com/dotnet/roslyn/pull/40145#discussion_r354033064) [](ancestors = 354033064)

Reply:
> No, because there is a conversion for assignment from null to System.Exception.

I think the adjusted test clearly demonstrates that ```throw null;``` is broken in the current language version, works in 7.0.

---
In reply to: [357854151](https://github.com/dotnet/roslyn/pull/40145#discussion_r357854151) [](ancestors = 357854151,354033064)

Reply:
You mean when there is no `Exception` type?  Sure, `throw null;` will no longer bind.

Reply:
> You mean when there is no Exception type?  Sure, throw null; will no longer bind.

I think compat council should approve this.

---
In reply to: [357870727](https://github.com/dotnet/roslyn/pull/40145#discussion_r357870727) [](ancestors = 357870727)

Reply:
And the breaking change documented

---
In reply to: [357871177](https://github.com/dotnet/roslyn/pull/40145#discussion_r357871177) [](ancestors = 357871177,357870727)